src/java.base/windows/classes/java/lang/ProcessImpl.java

```diff
@@ -505,39 +484,36 @@ private ProcessImpl(String cmd[],
 
         processHandle = ProcessHandleImpl.getInternal(getProcessId0(handle));
 
-        java.security.AccessController.doPrivileged(
-        new java.security.PrivilegedAction<Void>() {
-        public Void run() {
-            if (stdHandles[0] == -1L)
-                stdin_stream = ProcessBuilder.NullOutputStream.INSTANCE;
-            else {
-                FileDescriptor stdin_fd = new FileDescriptor();
-                fdAccess.setHandle(stdin_fd, stdHandles[0]);
-                fdAccess.registerCleanup(stdin_fd);
-                stdin_stream = new BufferedOutputStream(
-                    new PipeOutputStream(stdin_fd));
-            }
+        if (stdHandles[0] == -1L)
+            stdin_stream = ProcessBuilder.NullOutputStream.INSTANCE;
+        else {
+            FileDescriptor stdin_fd = new FileDescriptor();
+            fdAccess.setHandle(stdin_fd, stdHandles[0]);
+            fdAccess.registerCleanup(stdin_fd);
+            stdin_stream = new BufferedOutputStream(
+                new PipeOutputStream(stdin_fd));
+        }
 
-            if (stdHandles[1] == -1L || forceNullOutputStream)
-                stdout_stream = ProcessBuilder.NullInputStream.INSTANCE;
-            else {
-                FileDescriptor stdout_fd = new FileDescriptor();
-                fdAccess.setHandle(stdout_fd, stdHandles[1]);
-                fdAccess.registerCleanup(stdout_fd);
-                stdout_stream = new BufferedInputStream(
-                    new PipeInputStream(stdout_fd));
-            }
+        if (stdHandles[1] == -1L || forceNullOutputStream)
+            stdout_stream = ProcessBuilder.NullInputStream.INSTANCE;
+        else {
+            FileDescriptor stdout_fd = new FileDescriptor();
+            fdAccess.setHandle(stdout_fd, stdHandles[1]);
+            fdAccess.registerCleanup(stdout_fd);
+            stdout_stream = new BufferedInputStream(
+                new PipeInputStream(stdout_fd));
+        }
 
-            if (stdHandles[2] == -1L)
-                stderr_stream = ProcessBuilder.NullInputStream.INSTANCE;
-            else {
-                FileDescriptor stderr_fd = new FileDescriptor();
-                fdAccess.setHandle(stderr_fd, stdHandles[2]);
-                fdAccess.registerCleanup(stderr_fd);
-                stderr_stream = new PipeInputStream(stderr_fd);
-            }
+        if (stdHandles[2] == -1L)
+            stderr_stream = ProcessBuilder.NullInputStream.INSTANCE;
+        else {
+            FileDescriptor stderr_fd = new FileDescriptor();
+            fdAccess.setHandle(stderr_fd, stdHandles[2]);
+            fdAccess.registerCleanup(stderr_fd);
+            stderr_stream = new PipeInputStream(stderr_fd);
+        }
 
-            return null; }});
+        return null;
```

Comment:
I think this is an oversight - returning `null` from the constructor would be an compile error.

Reply:
Actually, looking at the GitHub actions job, it appears to have caught one other compilation error in this file:

```
jdk\jdk\src\java.base\windows\classes\java\lang\ProcessImpl.java:423: error: ';' expected
        final String value = System.getProperty("jdk.lang.Process.allowAmbiguousCommands", "true"));
                                                                                                  ^
1 error
```

Reply:
I assume Roger will run at least tier1 on all platforms before integrating.

Reply:
Inadvertent lapse.
Retested in the CI on all platforms

Reply:
Thank you Roger, the updated changes look good to me.